### ISSUE TITLE
Spark SQL Functions Coverage and Parity - Array

### ISSUE THREAD

**Issue by shehabgamin** [lakehq/sail#220]
- [x] array 
- [x] array_append
- [x] array_compact
- [x] array_contains
- [x] array_contains_all
- [x] array_distinct
- [x] array_except
- [x] array_insert
- [x] array_intersect
- [ ] array_join (Partial parity)
- [x] array_max
- [x] array_min
- [x] array_position
- [x] array_prepend
- [x] array_remove
- [x] array_repeat
- [x] array_union
- [x] arrays_overlap
- [x] arrays_zip
- [x] flatten
- [x] get
- [ ] sequence (Partial parity)
- [ ] shuffle
- [x] slice
- [x] sort_array

**Comment by keen85** [comment]
@shehabgamin, @SparkApplicationMaster 
I was not able to find anything about `array_contains_all` as [PySpark function](https://spark.apache.org/docs/latest/api/python/search.html?q=array_contains_all#) nor as [Spark SQL function](https://spark.apache.org/docs/latest/api/sql/search.html?q=array_contains_all).
So, is this some proprietary function of `Sail`?

**Comment by shehabgamin** [comment]
@keen85 You're right, not sure where I got that function from when creating this issue. Other engines seem to have this function but Spark doesn't.
